src/Adapter/Debug/DebugModeConfiguration.php

```diff
@@ -169,4 +187,38 @@ private function updateDebugMode($enableStatus)
 
         return null;
     }
+
+    /**
+     * Update module status
+     *
+     * @param bool $disableNonNativeModules
+     *
+     * @return void
+     */
+    private function updateModulesStatus(bool $disableNonNativeModules): void
```

Comment:
I wonder about this way of enabling modules Let me explain, when you set this option to false it makes sense to disable all non-native modules that one is ok

However, when you set it back to true all the non-native modules will be automatically re-enabled, but what if some of them were previously before regardless of their status (native or not) then they could be re-enabled while it was not expected which could be a problem

Reply:
So ideally maybe we would need an alternative way of disabling the non-native modules One that is not related to their `active` DB value I have a couple ideas:
- move the module's folder to another folders than `modules` similarly to what we started implementing in the autoupgrade module, when a module is not in the expected folder any more it is not executed therefore it can be considered as disabled (or even uninstalled) but all its DB data remains intact
- handle a special filter when the core gets the list of installed modules In `ModulesRepository::getActiveModules()` we could add special filtering in the DB query when the option is Disabled then we filter the non-native modules from the DB result Theoretically, if this list doesn't return the module then it is not executed either (same principle as the `active = 0` column)

If we do one of these solutions then it should have the same effect but it has two benefits:
- the former `active` state of a module is never lost
- we don't need to execute a command here, only update the configuration which should be (slightly) more optimized

Reply:
Note that if we choose one of these alternatives the CQRS command `BulkToggleModuleStatusCommand` might become useless and will need to be removed But it's not an obligation since it could have other uses! And it's already fully behat tested